### PR TITLE
Story/ccls-1998_2013_15 opponents endpoints

### PR DESCRIPTION
**Pull request by porritta**
Includes
CCLS-1998
CCLS-2013
CCLS-2014
CCLS-2015

### CODE REVIEW

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java [hunk 1]
```diff
@@ -250,43 +250,43 @@ public class Opponent implements Serializable {
...
   @Column(name = "CONFIRMED")
-  private boolean confirmed;
+  private Boolean confirmed;
 
   /**
    * Indicates the application mode.
    */
   @Column(name = "APPMODE")
-  private boolean appMode = true;
+  private Boolean appMode = true;
```
Comment (PhilDigitalJustice) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java:259]: Should we not be setting these?

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java [hunk 2]
```diff
@@ -250,43 +250,43 @@ public class Opponent implements Serializable {
...
 
   /**
    * Indicates if the Opponent represents an amendment.
    */
   @Column(name = "AMENDMENT")
-  private boolean amendment = false;
+  private Boolean amendment = false;
```
Comment (PhilDigitalJustice) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java:265]: Should we not be setting these?

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java [hunk 3]
```diff
@@ -250,43 +250,43 @@ public class Opponent implements Serializable {
...
 
   /**
    * Indicates if the Opponent is awarded.
    */
   @Column(name = "AWARD")
-  private boolean award = false;
+  private Boolean award = false;
```
Comment (PhilDigitalJustice) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java:271]: Should we not be setting these?

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java [hunk 4]
```diff
@@ -250,43 +250,43 @@ public class Opponent implements Serializable {
...
   @Column(name = "PUBLIC_FUNDING_APPLIED")
-  private boolean publicFundingApplied;
+  private Boolean publicFundingApplied;
 
   /**
    * Indicates if the Opponent is shared.
    */
   @Column(name = "SHARED_IND")
-  private boolean sharedInd = false;
+  private Boolean sharedInd = false;
```
Comment (PhilDigitalJustice) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java:283]: Should we not be setting these?

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java [hunk 5]
```diff
@@ -250,43 +250,43 @@ public class Opponent implements Serializable {
...
 
   /**
    * Indicates if the Opponent is marked for deletion.
    */
   @Column(name = "DELETE_IND")
-  private boolean deleteInd = true;
+  private Boolean deleteInd = true;
```
Comment (PhilDigitalJustice) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java:289]: Should we not be setting these?

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Proceeding.java [hunk 6]
```diff
@@ -181,7 +181,7 @@ public class Proceeding implements Serializable {
    * Lead proceeding indicator.
    */
   @Column(name = "LEAD_PROCEEDING_IND")
-  private boolean leadProceedingInd = false;
+  private Boolean leadProceedingInd = false;
```
Comment (PhilDigitalJustice) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Proceeding.java:184]: Should we not be setting these?

**Review by PhilDigitalJustice — CHANGES_REQUESTED** [review]
A few comments around the same issue, rest looks good

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java (hunk 1 above)
Comment (porritta) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java]: Amended

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java (hunk 3 above)
Comment (porritta) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java]: Amended

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java (hunk 5 above)
Comment (porritta) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Opponent.java]: Amended

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Proceeding.java (hunk 6 above)
Comment (porritta) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/Proceeding.java]: Amended

**Review by PhilDigitalJustice — APPROVED** [review]

**Comment by PhilDigitalJustice** [comment]
LGTM

**Pull request merged**